src/qopenglwebpage.cpp

```diff
@@ -186,6 +188,17 @@ void QOpenGLWebPage::drawUnderlay()
 */
 void QOpenGLWebPage::drawOverlay(const QRect &rect)
 {
+    QList<QWeakPointer<QMozGrabResult> >::const_iterator it = mGrabResultList.begin();
+    for (; it != mGrabResultList.end(); ++it) {
+        QSharedPointer<QMozGrabResult> result = it->toStrongRef();
+        if (result) {
+            result->captureImage(rect);
+        } else {
+            qWarning() << "QMozGrabResult freed before being realized!";
+        }
+    }
+    mGrabResultList.clear();
+
     Q_EMIT afterRendering(rect);
```

Comment:
Logically it is the afterRendering signal that should have been used for capturing. Meaning that direct connection should have been used.